app/models/concerns/episode_media.rb

```diff
@@ -7,8 +7,11 @@ def complete_media
     contents.complete_or_replaced.group_by(&:position).values.map(&:first)
   end
 
+  # TODO: not ideal, but we need to ensure this stays true forever, after the
+  # first time the episode is written to any feed. for now, just assume that
+  # takes about an hour.
```

Comment:
Long term plan here: add a `media_resource_versions` table to track episode `Content` that can be grouped together.  (And maybe only create these records when all contents in that version are "complete"?).

Basically, we want to track any past "version" of the audio we served.  Because the user may change the episode segment_count / medium / re-upload-files / re-cut-files, making the current files invalid.  And we need something to fall back on.

Reply:
That makes sense. Versioning the media with each mutation would be a great way to tackle these race conditions and start picking apart the details of the concurrency model.  Though mostly the details are handled already!

Reply:
How will adding versions subsequent to this patch fit with the changes here?  Where can I find a sketch of what that work consists of?

Reply:
Started working on that in [#816](https://github.com/PRX/feeder.prx.org/pull/816/files#diff-bda765f89a866286ad1f1b6a4cf2ed1eed4bb2b6859d7e354da995bfaf139f32R11).  Basically, whenever we have a set of completed files for every position in the segment_count ... we cut a new version.

(I'm still mulling over that branch ... was going for minimal changes on the first pass)